kyuubi-common/src/main/scala/org/apache/kyuubi/engine/EngineAppName.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine
+
+import java.net.InetAddress
+
+import org.apache.curator.utils.ZKPaths
+
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.config.KyuubiConf.{ENGINE_SCOPE, FRONTEND_BIND_HOST, FRONTEND_BIND_PORT}
+import org.apache.kyuubi.engine.EngineScope.{GROUP, SERVER, SESSION, USER}
+
+class EngineAppName(user: String, sessionId: String, conf: KyuubiConf) {
+
+  import EngineAppName._
+
+  private val engineScope = EngineScope.withName(conf.get(ENGINE_SCOPE))
+  private val serverHost = conf.get(FRONTEND_BIND_HOST)
+    .getOrElse(InetAddress.getLocalHost.getHostName)
+  private val serverPort = conf.get(FRONTEND_BIND_PORT)
```

Comment:
How about using the connection URL of the server after the port is actually resoved？

Reply:
this is the kyuubiServer's host and port ,It comes from the kyuubiServer configuration.

Reply:
we support 0 as a random port, which need to be resolved to actual port

Reply:
if the kyuubiServer use a random port, how does the client connect to the service

Reply:
say way how the server connect engines

Reply:
no, It's port number for Kyuubi, not engine. If we parse to the actual port number, then write back to the kyuubiServer 
 configuration.How about this？

Reply:
yes this is the port for kyuubi server and it supports 0 for a random pick, we have a HA layer that support dynamic service discovery

Reply:
So, I have found another issue for you, the zookeeper directories that you designed so far, may never consider the conflict with the server `znode`

Reply:
It‘s HA server `znode`?